test/testdrive/sinks.td

```diff
@@ -196,3 +196,26 @@ snk7        user
 snk8        user
 sink9       user
 sink10      user
+
+# test explicit partition count
```

Comment:
its maybe worth adding a test where the user explicitly requests partition_count and / or replication_factor = -1?

Reply:
Ah, good point, thank you! Added